src/execute_lua_script.cc

```diff
@@ -32,34 +32,49 @@
 
 namespace task {
 
-std::variant<sol::object, std::string>
-execute_lua_script(sol::state& lua, sol::string_view script)
+namespace {
+
+const std::string anchor = u8"\u2693";
+constexpr gul14::string_view chunk_prefix{ u8"[string \"\u2693\"]:" };
+
+std::string process_msg(gul14::string_view msg)
 {
-    const std::string anchor = u8"\u2693";
-    constexpr gul14::string_view chunk_prefix{ u8"[string \"\u2693\"]:" };
+    // If C++ code is called by Lua and throws an exception that is not derived
+    // from std::exception, the exception is not intercepted by the Sol
+    // trampoline, but caught directly by Lua. Lua would expect this exception
+    // to come from lua_error() and therefore looks for an error message on its
+    // stack, which is not there. Depending on build type and Sol2 configuration,
+    // this can generate a stack error message or not. We try to convert this into
+    // a concise error message.
+    if (msg.empty() ||
+        msg == "lua: error: stack index 1, expected string, received function")
+    {
+        return "Unknown exception";
+    }
 
-    auto process_msg =
-        [chunk_prefix](gul14::string_view msg) -> std::string
-        {
-            // If C++ code is called by Lua and throws an exception that is not derived
-            // from std::exception, the exception is not intercepted by the Sol
-            // trampoline, but caught directly by Lua. Lua would expect this exception
-            // to come from lua_error() and therefore looks for an error message on its
-            // stack, which is not there. Depending on build type and Sol2 configuration,
-            // this can generate a stack error message or not. We try to convert this into
-            // a concise error message.
-            if (msg.empty() ||
-                msg == "lua: error: stack index 1, expected string, received function")
-            {
-                return "Unknown exception";
-            }
-
-            return gul14::replace(msg, chunk_prefix, "");
-        };
+    return gul14::replace(msg, chunk_prefix, "");
+}
+
+} // anynomous namespace
 
+
+std::string check_lua_script(sol::state& lua, sol::string_view script)
+{
+    sol::load_result result = lua.load(script, anchor);
```

Comment:
Ah, ok: let Lua check for syntax errors :-) At least there is a forward to `lua_load` functionality. According to the [Lua documentation](https://www.lua.org/manual/5.4/manual.html#lua_load) we can have different results (see also section 4.1.1 [Status Codes](https://www.lua.org/manual/5.4/manual.html#4.4.1)). Should we forward them to the user without any further interpretation? I would suppose that this is ok.

Reply:
What we get from Sol2 in `what()` is already a pre-processed error message. It has the same format like the error messages we get from the `lua.safe_script()` call below.